src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/fs/PrependToFileStep.java

```diff
@@ -0,0 +1,160 @@
+import org.kohsuke.stapler.QueryParameter;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.Descriptor;
+import hudson.util.FormValidation;
+
+/**
+ * Prepend given content to a given file.
+ */
+public class PrependToFileStep extends Step {
+	private final String file;
+	private final String content;
+
+	@DataBoundConstructor
+	public PrependToFileStep(final String file, final String content) throws Descriptor.FormException {
+		if (StringUtils.isBlank(file)) {
+			throw new Descriptor.FormException("can't be blank", "file");
+		}
+		this.file = file;
+
+		if (StringUtils.isBlank(content)) {
+			throw new Descriptor.FormException("can't be blank", "content");
+		}
+		this.content = content;
+	}
+
+	/**
+	 * The file to prepend to.
+	 *
+	 * @return the file
+	 */
+	public String getFile() {
+		return this.file;
+	}
+
+	/**
+	 * Gets the content to prepend.
+	 *
+	 * @return the content
+	 */
+	public String getContent() {
+		return this.content;
+	}
+
+	@Override
+	public StepExecution start(final StepContext context) throws Exception {
+		return new ExecutionImpl(this, context);
+	}
+
+	@Extension
+	public static class DescriptorImpl extends StepDescriptor {
+
+		public DescriptorImpl() {
+
+		}
+
+		@Override
+		public Set<? extends Class<?>> getRequiredContext() {
+			return Collections.singleton(FilePath.class);
+		}
+
+		@Override
+		public String getFunctionName() {
+			return "prependToFile";
+		}
+
+		@Override
+		public String getDisplayName() {
+			return "Create a file (if not already exist) in the workspace, and prepend given content to that file.";
+		}
+
+		@SuppressWarnings("unused")
+		public FormValidation doCheckFile(@QueryParameter final String value) {
+			if (StringUtils.isBlank(value)) {
+				return FormValidation.error("Needs a value");
+			} else {
+				return FormValidation.ok();
+			}
+		}
+
+		@SuppressWarnings("unused")
+		public FormValidation doCheckContent(@QueryParameter final String value) {
+			if (StringUtils.isBlank(value)) {
+				return FormValidation.error("Needs a value");
+			} else {
+				return FormValidation.ok();
+			}
+		}
+	}
+
+	/**
+	 * The execution of {@link PrependToFileStep}.
+	 */
+	public static class ExecutionImpl extends SynchronousNonBlockingStepExecution<FileWrapper> {
+		private static final long serialVersionUID = 1L;
+
+		private transient PrependToFileStep step;
+
+		protected ExecutionImpl(@Nonnull final PrependToFileStep step, @Nonnull final StepContext context) {
+			super(context);
+			this.step = step;
+		}
+
+		@Override
+		protected FileWrapper run() throws Exception {
+			final FilePath ws = this.getContext().get(FilePath.class);
+			assert ws != null;
+			final FilePath file = ws.child(this.step.getFile());
+			if (!file.exists()) {
+				file.getParent().mkdirs();
+				file.touch(System.currentTimeMillis());
+			}
+			final String content = this.step.getContent();
+			final String originalContent = file.readToString();
```

Comment:
A future optimization to do on this would be to have everything run on the agent in a `MasterToSlaveFileCallable` so that we aren't sending big strings back and forth between the agent and controller as well as not using up memory on the controller for temporarily holding those strings.